### PR TITLE
fix/338: add Util function to fetch primary category

### LINKED ISSUE #338: Update the mapping for section_names field to reflect primary category
### Is your enhancement related to a problem? Please describe.

Currently both the `section_names` and `all_sections` field displays the same information. The reason why `section_names` is displaying all section information is because of a work around we had to do when `all_sections` field was not available. Since all_section is available right now, we can go ahead and update the mapping for `section_names` field to only reflect primary category.

### Designs

_No response_

### Describe alternatives you've considered

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Sidsector9**
### Description of the Change

Adds a Util function `get_primary_category`.

This function returns the primary category name set through Yoast.
If Yoast is deactivated, then it returns the category set at index 0.

Closes #338 

### Checklist:

- [ ] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Credits

Props @vjayaseelan90 @jeffpaul @Sidsector9

### CODE REVIEW

File: tests/phpunit/test-tools/Utils_Tests.php
Comment (cadic) [inline tests/phpunit/test-tools/Utils_Tests.php]: Looks like this last test is a full-copy of the previous one (except the category name) and does not test any alternative workflow.

**Review by cadic — CHANGES_REQUESTED** [review]
The code looks good. I have a concern about the 4th Unit test added.

File: tests/phpunit/test-tools/Utils_Tests.php
Comment (Sidsector9) [reply tests/phpunit/test-tools/Utils_Tests.php]: you're correct!

**Review by cadic — APPROVED** [review]

**Comment by jeffpaul** [comment]
Holding off on merge here as we're awaiting Sophi team approval on the pending 1.3.0 release and I'd prefer to not change any additional code.  Once 1.3.0 is released, I'll merge this in so we can deploy/test and prepare as part of a 1.3.1/1.4.0 release.

**Pull request merged**
